src/resources/basis.js

```diff
@@ -1,518 +1,311 @@
+        });
+    };
 
-            if (!basisFile.transcodeImage(dst, 0, mip, basisFormat, 1, 0)) {
-                basisFile.close();
-                basisFile.delete();
-                throw new Error('Failed to transcode image url=' + url);
-            }
+    if (config.glueUrl && config.wasmUrl && wasmSupported()) {
+        let basisCode = null;
+        let module = null;
 
-            if (basisFormat === BASIS_FORMAT.cTFRGB565 || basisFormat === BASIS_FORMAT.cTFRGBA4444) {
-                // 16 bit formats require Uint16 typed array
-                const dst16 = new Uint16Array(dstSize / 2);
-                for (i = 0; i < dstSize / 2; ++i) {
-                    dst16[i] = dst[i * 2] + dst[i * 2 + 1] * 256;
+        // download glue script
+        http.get(config.glueUrl, { responseType: 'blob' }, (err, response) => {
+            if (err) {
+                callback(err);
+            } else {
+                if (module) {
+                    sendResponse(response, module);
+                } else {
+                    basisCode = response;
                 }
-                dst = dst16;
             }
+        });
 
-            levelData.push(dst);
-        }
-
-        basisFile.close();
-        basisFile.delete();
-
-        // handle unswizzle option
-        if (options && options.unswizzleGGGR) {
-            basisFormat = BASIS_FORMAT.cTFRGB565;
-            for (i = 0; i < levelData.length; ++i) {
-                levelData[i] = pack565(unswizzleGGGR(levelData[i]));
-            }
-        }
-
-        return {
-            format: basisToEngineMapping[basisFormat],
-            width: (width + 3) & ~3,
-            height: (height + 3) & ~3,
-            levels: levelData,
-            cubemap: false,
-            mipmaps: true,
-            transcodeTime: hasPerformance ? (performance.now() - funcStart) : 0,
-            url: url
-        };
-    };
-
-    let basis = null;
-    let queue = [];
-
-    // download and transcode the file given the basis module and
-    // file url
-    const workerTranscode = function (url, format, data, options) {
-        try {
-            // texture data has been provided
-            const result = transcode(basis, url, format, data, options);
-            result.levels = result.levels.map(function (v) {
-                return v.buffer;
-            });
-            self.postMessage({ url: url, data: result }, result.levels);
-        } catch (err) {
-            self.postMessage({ url: url.toString(), err: err.toString() });
-        }
-    };
-
-    // Initialize the web worker's BASIS interface.
-    // basisModule is the optional wasm binary.
-    const workerInit = function (basisModule) {
-        const instantiateWasmFunc = function (imports, successCallback) {
-            WebAssembly.instantiate(basisModule, imports)
-                .then(function (result) {
-                    successCallback(result);
+        const fetchPromise = fetch(config.wasmUrl);
```

Comment:
Just wanted to flag that the updated code will call `fetch` if WASM is supported whereas before it was just if compile streaming is supported. I think it's OK though since `fetch` _should_ be supported everywhere where WASM is supported.